### PR TITLE
[core] Add file compression type in file names

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Default, the file name written by paimon only has the file format, it can also include the file compression like spark.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/io/DataFilePathFactory.java [hunk 1]
```diff
@@ -62,7 +65,8 @@ public Path newChangelogPath() {
     }
 
     private Path newPath(String prefix) {
-        String name = prefix + uuid + "-" + pathCount.getAndIncrement() + "." + formatIdentifier;
+        String extension = "." + fileCompression + "." + formatIdentifier;
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/io/DataFilePathFactory.java:68]: Please add a config option that whether add fileCompression or not.

**Comment by wwj6591812** [comment]
Why you submit this pr?
What are the business requirements for this feature?

**Comment by askwang** [comment]
Has added a config option. thanks for review @JingsongLi @wwj6591812

**Comment by askwang** [comment]
> Why you submit this pr? What are the business requirements for this feature?

It has no business effect, just provide convenience to know the compression type based on the written file name.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**